db/migrate/20231018192110_add_index_to_webauthn_credentials_user_id_nickname.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class AddIndexToWebauthnCredentialsUserIdNickname < ActiveRecord::Migration[7.0]
+  disable_ddl_transaction!
+
+  def up
+    safety_assured do
+      execute <<~SQL.squish
+        DELETE FROM webauthn_credentials
+          WHERE id NOT IN (
+          SELECT DISTINCT ON(nickname, user_id) id FROM webauthn_credentials
+          ORDER BY nickname, user_id, id ASC
+        )
+      SQL
+    end
+
+    add_index :webauthn_credentials, [:nickname, :user_id], unique: true, algorithm: :concurrently
```

Comment:
Race conditions between the two statements, while very unlikely, could cause the second statement to fail with a dangling invalid index.

I wonder if we shouldn't rather use the pattern I used in `db/post_migrate/20230803082451_add_unique_index_on_preview_cards_statuses.rb` (no need for the `ctid` stuff though, the `DISTINCT` query you already have should do the trick)

Reply:
Pushed a change which adds the retry loop with reindexing when recordnotunique is caught.